integration_tests/test_aggregator_settlement.py

```diff
@@ -0,0 +1,174 @@
+                                                       energy_forecast_kWh={
+                                                           forecast_market_slot: energy_forecast
+                                                       }) \
+            .set_energy_measurement(asset_uuid=area_uuid,
+                                    energy_measurement_kWh={
+                                        current_market_slot: energy_measurement
+                                    })
+
+    @staticmethod
+    def _can_place_in_settlement_market(asset_info):
+        return "unsettled_deviation_kWh" in asset_info and (
+                asset_info["unsettled_deviation_kWh"] != 0)
+
+    def on_market_cycle(self, market_info):
+        logging.info("market_info: %s", market_info)
+
+        try:
+            for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+                if not area_dict.get("asset_info"):
+                    continue
+
+                # manage forecast and measurement information
+
+                current_market_slot = market_info["market_slot"]
+                forecast_market_slot = self._next_market_slot(market_info)
+
+                if self._can_place_forecast_measurements(area_dict):
+                    self._manage_forecasts_measurements(area_uuid, area_dict,
+                                                        forecast_market_slot,
+                                                        current_market_slot)
+
+                    response = self.send_batch_commands()
+                    logging.info("forecast/measurement placed on the new market: %s", response)
+
+                # manage bids and offers in spot market
+
+                asset_info = area_dict["asset_info"]
+                if self._can_place_offer(asset_info):
+                    energy = asset_info["available_energy_kWh"] / 2
+                    self.add_to_batch_commands.offer_energy(asset_uuid=area_uuid, price=1,
+                                                            energy=energy)
+                if self._can_place_bid(asset_info):
+                    energy = asset_info["energy_requirement_kWh"] / 2
+                    self.add_to_batch_commands.bid_energy(asset_uuid=area_uuid, price=30,
+                                                          energy=energy)
+
+                response = self.send_batch_commands()
+                if response:
+                    logging.info("bid/offer placed on the spot market: %s", response)
+
+                # manage bids and offers in settlement markets
+
+                if self._can_place_in_settlement_market(asset_info):
+
+                    for market_slot, energy in asset_info["unsettled_deviation_kWh"].items():
+
+                        if energy and energy > 0:
+                            self.add_to_batch_commands.bid_energy(asset_uuid=area_uuid,
+                                                                  energy=energy / 2, price=30,
```

Comment:
why are you only sending half of the energy here? I think it would be a good idea bid all the energy in order to test the correct exposing of the unsettled energy.

Reply:
There was no reason other than I base on other test. Comment addressed here [7b5d0d7](https://github.com/gridsingularity/gsy-e-sdk/pull/207/commits/7b5d0d7ed599e94a7ebdf622b816d7d5e7453c05). I also added a test for what you mention.